### ISSUE TITLE
[sklearn#34628] Support custom per-class decision thresholds and a rejection option in `sklearn.metrics.confusion_matrix` and `sklearn.metrics.ConfusionMatrixDisplay`.

### ISSUE THREAD

**Issue by godofecht** [flooooooooooow/flow#306]
**Mirrored from [scikit-learn#34628](https://github.com/scikit-learn/scikit-learn/issues/34628)**

Original title: Support custom per-class decision thresholds and a rejection option in `sklearn.metrics.confusion_matrix` and `sklearn.metrics.ConfusionMatrixDisplay`.
Original labels: New Feature, Needs Triage
Created: 2026-08-01T18:23:17Z

---

> [!WARNING]
> This issue is not yet ready for a PR. If you are interested in contributing to scikit-learn, please have a look at our [contributing guidelines](https://scikit-learn.org/dev/developers/contributing.html), and in particular the sections for [new contributors](https://scikit-learn.org/dev/developers/contributing.html#new-contributors) and the ["Needs triage"](https://scikit-learn.org/dev/developers/contributing.html#issues-tagged-needs-triage) label.

### Describe the workflow you want to enable

Currently, in the code, the confusion matrix implementation relies on discrete predictions of y_pred on the basis of either 'argmax' for multi-class or a scalar cutoff value. 
In many real-world, risk-sensitive machine learning workflows (e.g., fraud detection, medical diagnostics, or financial compliance), models operate under a selective classification paradigm ("classification with a rejection option"). 

### Describe your proposed solution

With this issue, I propose a custom parameter 'thresholds' to 'confusion_matrix' and it's visual counterparts that are ConfusionMatrixDisplay 'from_estimator', and 'from_predictions' that accepts a custom decision boundary for per class label prediction , along side the 'y_score'.

Parameter:
thresholds can be an array-like of shape (n_classes,) or a dict, with a default of None. It accepts an array of threshold floats corresponding to target classes or a dictionary mapping class labels to float thresholds. When thresholds is provided, the function accepts 2D probability or score arrays (y_score) of shape (n_samples, n_classes).

Rejection region:
For any sample that is rejected out of any class label, expand the matrix by n+1 , or append a rejection column for predictions. Display labels dynamically update to include 'Unassigned' for string or categorical targets, or -1 for integer targets.

 

### Describe alternatives you've considered, if relevant

None

---

## Flow context

This issue is mirrored from scikit-learn to explore whether Flow encounters the same problem or can approach it differently. Add analysis below.

**Comment by godofecht** [comment]
Moved to https://github.com/godofecht/flow-scikit/issues/117 in the flow-scikit repository. Further library-specific triage and implementation will happen there.
